File: android/src/main/kotlin/co/quis/flutter_contacts/properties/Account.kt
package co.quis.flutter_contacts.properties

data class Account(
    var rawId: String,
    var type: String,
    var name: String,
    var mimetypes: List<String> = listOf<String>()
) {

    companion object {
        fun fromMap(m: Map<String, Any>): Account = Account(
            m["rawId"] as String,
            m["type"] as String,
            m["name"] as String,
            m["mimetypes"] as List<String>
        )
    }

    fun toMap(): Map<String, Any> = mapOf(
        "rawId" to rawId,
        "type" to type,
        "name" to name,
        "mimetypes" to mimetypes
    )
}

File: darwin/Classes/properties/Account.swift
import Contacts

@available(iOS 9.0, *)
struct Account {
    var rawId: String
    // local, exchange, cardDAV or unassigned
    var type: String
    var name: String

    init(fromMap m: [String: Any]) {
        rawId = m["rawId"] as! String
        type = m["type"] as! String
        name = m["name"] as! String
    }

    init(fromContainer c: CNContainer) {
        rawId = c.identifier
        name = c.name
        switch c.type {
        case .local:
            type = "local"
        case .exchange:
            type = "exchange"
        case .cardDAV:
            type = "cardDAV"
        case .unassigned:
            type = "unassigned"
        default:
            type = "unassigned"
        }
    }

    func toMap() -> [String: Any] { [
        "rawId": rawId,
        "type": type,
        "name": name,
        "mimetypes": [String](), // not available on iOS
    ]
    }
}
